server/app/filters/ValidAccountFilter.java

```diff
@@ -42,6 +45,22 @@ public EssentialAction apply(EssentialAction next) {
         });
   }
 
+  private boolean shouldLogoutUser(CiviFormProfile profile) {
+    return !profileUtils.validCiviFormProfile(profile) || !isValidSession(profile);
+  }
+
+  private boolean isValidSession(CiviFormProfile profile) {
+    if (config.getBoolean("session_replay_protection_enabled")) {
+      profile
+          .getAccount()
+          .thenApply(
+              account ->
+                  account.getActiveSession(profile.getProfileData().getSessionId()).isPresent())
+          .join();
```

Comment:
(blocking) Shouldn't this have a return statement?

Reply:
Not sure how that happened. Fixed.